### PR TITLE
Add and remove features

### PR DESCRIPTION
**Pull request by aima98**
- Removing all hardcoded items from the tasks array.
- Creating a new JavaScript file for the new functionality.
- Implementing a function for adding a new task (add a new element to the array).
- Implementing a function for deleting a task (remove an element from the array).
- Implementing a function for editing task descriptions.
- By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique
- All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/modules/Delete.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  allTasks = localStorage.setItem('todo', JSON.stringify(allTasks));
+  btn.previousSibling.parentElement.parentElement.remove();
+
+  if (allTasks.length === 0) {
+    listDisplay.innerHTML = '<hr/><p>No tasks are available now!</p>';
+  }
+};
+
+export default deleteTask;
```
Comment (CollinsTatang) [inline src/modules/Delete.js:19]: Hi @aima98 
You have done a great job implementing the different features of this app but there is a little update to complete the task.

- [ ] Please your app throws an error every time a user clicks the side button to delete a to-do in the list.
**Take A Look Blow**
delete
**Please fix the above error**

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @aima98, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.

- [ ] Kindly check your README file and correct the duplication of the `Show your support` section and remember the `Acknowledgments` is placed just above the `License` section according to the [README](https://github.com/microverseinc/readme-template) format.
readme


_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 2]
```diff
@@ -23,15 +24,16 @@ window.addEventListener('DOMContentLoaded', () => {
...
     target.style.display = 'none';
-    deleteTask(Number(target.id));
+    deletion.deleteTask(Number(target.id));
     localStorage.setItem('todo', JSON.stringify(listDisplay));
   }
 
   if (trash) {
-    deleteTask(Number(trash.id));
+    deletion.deleteTask(Number(trash.id));
   }
```
Comment (Ibizugbe) [inline src/index.js]: Nice job so far👍 But you still haven't completely implemented the changes requested by the previous reviewer🥲 Your app still has some issues when calling the remove function🥲 Kindly take a look at the video below for clarity to see the current behavior of your app and make the necessary change👍



https://user-images.githubusercontent.com/64643995/228900081-4b81b271-7efb-4788-98f2-837da1e80f8b.mp4

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @aima98 👋👋👋, 

tryagain


Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights✅✅

- [x] Github flow practice has been used👍
- [x] add function works properly👍


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [OPTIONAL] Some sections are missing in your readme😉 Kindly note that we are required to make use of the [microverse readme template](https://github.com/microverseinc/readme-template) for our projects and every section must be correctly filled except the FAQ section which is optional😉 Kindly ensure your readme is properly documented to give your prooject a more professional look😉😉👍

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — COMMENTED** [review]
Hi @aima98, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that have a live demo section in your README file and to have a professional README file you should remove this section till you have your app deployed so you can just add the live demo section in case you have deployed link for your project is present. .✅❗💯🤩✨✨


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @aima98, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of GitHub flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**